### PR TITLE
respect gamerule for destruction

### PR DESCRIPTION
**Pull request by liach**

### CODE REVIEW

**Comment by liach** [comment]
Also respected events and outfitted tracks

**Comment by liach** [comment]
Should I include #1223 too?

**Comment by CovertJaguar** [comment]
I was planning to rewrite it using the Timberjack block scanning algorithm.

File: src/main/java/mods/railcraft/common/items/ItemCrowbar.java [hunk 1]
```diff
@@ -235,49 +239,53 @@ public void addInformation(ItemStack stack, EntityPlayer player, List<String> in
...
-        InvTools.dropItems(drops, world, pos);
-        world.setBlockToAir(pos);
+    private void removeAndDrop(World world, BlockPos pos, IBlockState state, EntityPlayer player) {
+        if (MinecraftForge.EVENT_BUS.post(new BlockEvent.BreakEvent(world, pos, state, player)))
+            return;
+        state.getBlock().removedByPlayer(state, world, pos, player, true);
+        if (world.getGameRules().getBoolean("doTileDrops")) {
+            List<ItemStack> drops = state.getBlock().getDrops(world, pos, state, 0);
+            InvTools.dropItems(drops, world, pos);
+        }
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/items/ItemCrowbar.java]: We should make this into a function in WorldPlugin.

**Review by CovertJaguar — CHANGES_REQUESTED** [review]

File: src/main/java/mods/railcraft/common/items/ItemCrowbar.java (hunk 1 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/items/ItemCrowbar.java]: Sounds good...

**Comment by liach** [comment]
ref #872

**Comment by liach** [comment]
@CovertJaguar Changed.

**Review by CovertJaguar — APPROVED** [review]

**Pull request merged**